src/API/Google/AdsCampaign.php

```diff
@@ -127,10 +131,16 @@ public function get_campaign( int $id ): array {
 		} catch ( ApiException $e ) {
 			do_action( 'woocommerce_gla_ads_client_exception', $e, __METHOD__ );
 
-			throw new Exception(
+			$errors = $this->get_api_exception_errors( $e );
+			throw new ExceptionWithResponseData(
 				/* translators: %s Error message */
-				sprintf( __( 'Error retrieving campaign: %s', 'google-listings-and-ads' ), $e->getBasicMessage() ),
-				$this->map_grpc_code_to_http_status_code( $e )
+				sprintf( __( 'Error retrieving campaign: %s', 'google-listings-and-ads' ), reset( $errors ) ),
+				$this->map_grpc_code_to_http_status_code( $e ),
+				null,
+				[
+					'errors' => $errors,
+					'id'     => $id,
```

Comment:
💅  WDYT about changing `id` to `campaignId` 

If we check the error structure provided in the PR description

```
{
	"message": "Error editing campaign: A money amount was less than the minimum CPC for currency.",
	"errors": {
		"MONEY_AMOUNT_LESS_THAN_CURRENCY_MINIMUM_CPC": "A money amount was less than the minimum CPC for currency.",
		"INVALID_ARGUMENT": "Request contains an invalid argument."
	},
	"id": 1234567890
}
```

I thought that the `id` prop was kinda `errorCode` or `errorId` which is pretty commonly added in API exceptions besides the message.  Then after reading the code I noticed it was actually a Campaign ID

Reply:
I wouldn't mind changing it, but WP and WC rest API's are generally in snake case. Also they pretty consistently use the field "id" regardless of whether it's a product or a category that's being addressed. All these requests are to the "campaigns" endpoint. So to me it would seem more inconsistent if we pass a specific "id" in the request but then in the error it would return a "campaignId" and not the same "id".

Reply:
> WP and WC rest API's are generally in snake case

Sure, was just as a name reference. It would be `campaign_id`, `error_id` etc. 

> Also they pretty consistently use the field "id" regardless of whether it's a product or a category that's being addressed. All these requests are to the "campaigns" endpoint.

I don't think this is a good approach. Since the response data is at the same level as the error data it could be a collision and misunderstanding of what 'id' means. So I don't think is a problem to change the status quo for this. 

For example, calling the API endpoint for `get_campaigns` and  getting 

```
{
	"message": "Error editing campaign: A money amount was less than the minimum CPC for currency.",
	"errors": {
		"MONEY_AMOUNT_LESS_THAN_CURRENCY_MINIMUM_CPC": "A money amount was less than the minimum CPC for currency.",
		"INVALID_ARGUMENT": "Request contains an invalid argument."
	},
	"id": 1234567890
}
``` 


Doesn't imply that `id` is related to Campaign ID. As mentioned, could be interpreted as `error_id` or something like that. On the other side,  `campaign_id` is defining the context by itself in a much clearer way than just `id`. 

> All these requests are to the "campaigns" endpoint. So to me it would seem more inconsistent if we pass a specific "id" in the request but then in the error it would return a "campaignId" and not the same "id".

Maybe a refactoring of `get_campaing` could help, I mean changing `$id` to `$campaign_id`? 

I see that in `edit_campaign` as well as `delete_campaing` functions we're using `$campaign_id` instead of just `$id` as a parameter. Makes more sense for me `campaign_id` as well in response there and refactoring `get_campaing` to use `$campaign_id`.  

WDYT?

Reply:
Okok, thanks for elaborating on this. Since it looks like is a lot of work changing it, I'd agree with you on leaving it like this.  In the case in future is creating problems we have always the option of opening a PR for doing the change.